src/sentry/api/bases/project.py

```diff
@@ -17,7 +18,15 @@ class ProjectPermission(TeamPermission):
     }
 
     def has_object_permission(self, request, view, project):
-        return super(ProjectPermission, self).has_object_permission(request, view, project.team)
+        result = super(ProjectPermission,
+                       self).has_object_permission(request, view, project.organization)
+
+        if not result:
+            return result
+
+        return any([
```

Comment:
nit: the `[]` bit isn't needed. You can just do `return any(has_team_permission(request, team, self.scope_map) for team in project.teams.all())`

There's also value in that wihtout the `[]` it's lazily evaluated as a generator and will return when the first one is truthy.